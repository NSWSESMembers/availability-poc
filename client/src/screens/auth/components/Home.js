import React from 'react';
import { Text } from 'react-native';
import PropTypes from 'prop-types';

import Icon from 'react-native-vector-icons/FontAwesome';
import { Button } from '../../../components/Button';
import { Center, Container, Footer, Half, Holder } from '../../../components/Container';

const Home = ({ version, codePushHash, onPressSignIn, onPressOAuth, onPressRegister }) => (
  <Container isAlt>
    <Center>
      <Icon name="bullhorn" size={90} color="#FFF" />
      <Text style={{ fontSize: 45, color: '#FFF' }}>Callout</Text>
      <Text style={{ fontSize: 15, color: '#FFF' }}>
      Version {version} Revision {codePushHash}
      </Text>
    </Center>
    <Footer>
      <Half>
        <Holder>
          <Button text="OAuth" onPress={onPressOAuth} />
        </Holder>
      </Half>
      <Half>
        <Holder>
          <Button text="Sign In" onPress={onPressSignIn} />
        </Holder>
      </Half>
      <Half>
        <Holder>
          <Button text="Register" onPress={onPressRegister} type="secondary" />
        </Holder>
      </Half>
    </Footer>
  </Container>
);
Home.propTypes = {
  version: PropTypes.string,
  codePushHash: PropTypes.string,
  onPressSignIn: PropTypes.func,
  onPressOAuth: PropTypes.func,
  onPressRegister: PropTypes.func,
};

export default Home;
