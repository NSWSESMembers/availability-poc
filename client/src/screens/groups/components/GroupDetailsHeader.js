import React from 'react';
import { View, Text } from 'react-native';
import PropTypes from 'prop-types';

import styles from './styles';
import IconButton from './IconButton';
import Icon from '../../../components/Icon';


const GroupDetailsHeader = ({
  groupIcon,
  groupName,
  tags,
  memberAlready,
  leaveGroup,
  joinGroup,
}) => (
  <View style={styles.topContainer}>
    <View style={styles.iconLeftHolder}>
      <Icon style={styles.icon} name={groupIcon} size={30} />
    </View>
    <View style={styles.textContainer}>
      <Text style={styles.titleTextBold} numberOfLines={1}>
        {groupName}
      </Text>
      <Text style={styles.subtitleText} numberOfLines={1}>
        {tags !== '' ? tags : 'No Tags'}
      </Text>
    </View>
    <IconButton
      onPress={memberAlready ? leaveGroup : joinGroup}
      icon={memberAlready ? 'fa-sign-out' : 'fa-sign-in'}
      text={memberAlready ? 'Leave Group' : 'Join Group'}
    />
  </View>
);

GroupDetailsHeader.propTypes = {
  groupIcon: PropTypes.string.isRequired,
  groupName: PropTypes.string.isRequired,
  tags: PropTypes.string.isRequired,
  memberAlready: PropTypes.bool.isRequired,
  leaveGroup: PropTypes.func.isRequired,
  joinGroup: PropTypes.func.isRequired,
};

export default GroupDetailsHeader;
