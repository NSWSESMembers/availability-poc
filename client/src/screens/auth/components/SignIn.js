import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { Container, Holder } from '../../../components/Container';
import { Input } from '../../../components/Input';
import { Button } from '../../../components/Button';
import { Progress } from '../../../components/Progress';

class SignIn extends Component {
  state = {
    username: '',
    password: '',
  }

  onChangeUsername = (value) => {
    this.setState({
      username: value,
    });
  };

  onChangePassword = (value) => {
    this.setState({
      password: value,
    });
  };

  onSignIn = () => {
    const { username, password } = this.state;
    this.props.onSignIn({ username, password });
  }

  render() {
    const { loading } = this.props;
    const { username, password } = this.state;
    return (
      <Container>
        <Holder margin transparent>
          <Input
            type="username"
            placeholder="Username"
            value={username}
            onChangeText={this.onChangeUsername}
          />
          <Input
            type="password"
            placeholder="Password"
            value={password}
            onChangeText={this.onChangePassword}
          />
          <Button text="Sign In" onPress={this.onSignIn} disabled={loading} />
          {loading && (
            <Container>
              <Progress />
            </Container>
          )}
        </Holder>
      </Container>
    );
  }
}

SignIn.propTypes = {
  onSignIn: PropTypes.func.isRequired,
  loading: PropTypes.bool,
};

export default SignIn;
