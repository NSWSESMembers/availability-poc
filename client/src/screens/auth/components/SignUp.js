import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { Container, Holder } from '../../../components/Container';
import { Input } from '../../../components/Input';
import { Button } from '../../../components/Button';
import { Progress } from '../../../components/Progress';

class SignUp extends Component {
  state = {
    username: '',
    password: '',
    email: '',
  };

  onChangeUsername = (value) => {
    this.setState({
      username: value,
    });
  };

  onChangeEmail = (value) => {
    this.setState({
      email: value,
    });
  };

  onChangePassword = (value) => {
    this.setState({
      password: value,
    });
  };

  onSignUp = () => {
    const { username, email, password } = this.state;
    this.props.onSignUp({ username, email, password });
  }

  render() {
    const { loading } = this.props;
    const { username, email, password } = this.state;
    return (
      <Container>
        <Holder margin transparent>
          <Input
            type="username"
            placeholder="Username"
            value={username}
            onChangeText={this.onChangeUsername}
          />
          <Input
            type="password"
            placeholder="Password"
            value={password}
            onChangeText={this.onChangePassword}
          />
          <Input
            type="email"
            placeholder="Email"
            value={email}
            onChangeText={this.onChangeEmail}
          />
          <Button text="Register" onPress={this.onSignUp} disabled={loading} />
        </Holder>
        {loading && (
          <Container>
            <Progress />
          </Container>
        )}
      </Container>
    );
  }
}

SignUp.propTypes = {
  onSignUp: PropTypes.func.isRequired,
  loading: PropTypes.bool,
};

export default SignUp;
