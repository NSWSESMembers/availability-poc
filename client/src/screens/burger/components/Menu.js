import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { FlatList } from 'react-native';

import { Container } from '../../../components/Container';
import { ListItem } from '../../../components/List';

class Menu extends Component {
  renderItem = ({ item }) => (
    <ListItem
      bold
      {...item}
    />
  )

  render() {
    const { props } = this;
    const items = [
      {
        title: 'User Profile',
        subtitle: 'Modify your display name and preferences',
        iconLeft: 'user-circle',
        onPress: props.onShowUserProfile,
      },
      {
        title: 'Force location update',
        subtitle: 'Test the location tracking system',
        iconLeft: 'compass',
        onPress: props.onUpdateLocation,
      },
      {
        title: 'Check for updates',
        subtitle: 'Install a new version of the app, if available',
        iconLeft: 'download',
        onPress: props.onCheckForUpdate,
      },
      {
        title: 'Submit feedback',
        subtitle: 'Please post in the Facebook feedback group',
        iconLeft: 'comments',
        onPress: props.onSubmitFeedback,
      },
      {
        title: 'Test Bugsnag',
        subtitle: 'Submit a mock bug report to Bugsnag',
        iconLeft: 'bug',
        onPress: props.onTestBugsnag,
      },
      {
        title: 'Internal Parameters',
        subtitle: 'Display the app version and other details',
        iconLeft: 'info-circle',
        onPress: props.onShowParams,
      },
      {
        title: 'Logout',
        subtitle: 'Sign out of this device',
        iconLeft: 'times-circle',
        onPress: props.onLogout,
      },
    ];

    return (
      <Container>
        <FlatList
          data={items}
          keyExtractor={item => item.title}
          renderItem={this.renderItem}
        />
      </Container>
    );
  }
}

Menu.propTypes = {
  onShowUserProfile: PropTypes.func.isRequired,
  onUpdateLocation: PropTypes.func.isRequired,
  onCheckForUpdate: PropTypes.func.isRequired,
  onSubmitFeedback: PropTypes.func.isRequired,
  onTestBugsnag: PropTypes.func.isRequired,
  onShowParams: PropTypes.func.isRequired,
  onLogout: PropTypes.func.isRequired,
};

export default Menu;
