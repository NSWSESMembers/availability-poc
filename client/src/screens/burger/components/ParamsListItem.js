import PropTypes from 'prop-types';
import React from 'react';
import { Clipboard, ToastAndroid } from 'react-native';

import { ListItem } from '../../../components/List';


const ParamsListItem = ({ title, detail }) => {
  const onPress = () => {
    Clipboard.setString(detail);
    ToastAndroid.show('Copied to clipboard', ToastAndroid.SHORT);
  };

  return (
    <ListItem
      title={title}
      subtitle={detail}
      onPress={onPress}
    />
  );
};

ParamsListItem.propTypes = {
  title: PropTypes.string,
  detail: PropTypes.string,
};

export default ParamsListItem;
