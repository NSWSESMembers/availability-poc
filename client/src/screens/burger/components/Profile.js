import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { ActivityIndicator, Text, View, Image } from 'react-native';
import Prompt from 'react-native-prompt';

import gravatar from '../../../selectors/gravatar';
import { Container } from '../../../components/Container';

const styles = {
  username: {
    borderColor: '#777',
    borderBottomWidth: 1,
    borderTopWidth: 1,
    paddingHorizontal: 8,
    paddingBottom: 2,
    paddingTop: 5,
    fontSize: 14,
  },
  usernameHeader: {
    backgroundColor: '#dbdbdb',
    color: '#000000',
    paddingHorizontal: 8,
    paddingBottom: 2,
    paddingTop: 5,
    fontSize: 16,
  },
  email: {
    borderColor: '#777',
    borderBottomWidth: 1,
    borderTopWidth: 1,
    paddingHorizontal: 8,
    paddingBottom: 2,
    paddingTop: 5,
    fontSize: 14,
  },
  emailHeader: {
    backgroundColor: '#dbdbdb',
    color: '#000000',
    paddingHorizontal: 8,
    paddingBottom: 2,
    paddingTop: 5,
    fontSize: 16,
  },
  imageContainer: {
    paddingRight: 20,
    alignItems: 'center',
    paddingHorizontal: 16,
    paddingVertical: 8,
  },
  inputBorder: {
    flexDirection: 'row',
    borderColor: '#dbdbdb',
    borderBottomWidth: 1,
    borderTopWidth: 1,
    paddingVertical: 8,
    backgroundColor: '#ed3434',
  },
  inputInstructions: {
    color: '#777',
    fontSize: 26,
    flex: 1,
  },
  userContainer: {
    paddingLeft: 16,
    backgroundColor: '#c6c0c0',
    paddingHorizontal: 16,
    paddingVertical: 8,
  },
  gravatar: {
    paddingHorizontal: 10,
  },
  userInner: {
    flexDirection: 'row',
    backgroundColor: '#dbdbdb',
    alignItems: 'center',
    paddingVertical: 8,
    paddingRight: 0,
  },
};

class Profile extends Component {
  state = {
    promptVisible: false,
  }

  updateDisplayName = (value) => {
    this.setState({ promptVisible: false });
    this.props.onChangeName(value);
  }

  openPrompt = () => {
    this.setState({ promptVisible: true });
  };

  closePrompt = () => {
    this.setState({ promptVisible: false });
  };

  render() {
    const { loading, user } = this.props;

    // render loading placeholder while we fetch data
    if (loading || !user) {
      return (
        <View style={[styles.loading, styles.container]}>
          <ActivityIndicator />
        </View>
      );
    }

    return (
      <Container>
        <Prompt
          title="Change Display Name"
          placeholder={user.displayName}
          defaultValue={user.displayName}
          visible={this.state.promptVisible}
          onCancel={this.closePrompt}
          onSubmit={this.updateDisplayName}
        />
        <View style={styles.userContainer}>
          <View style={styles.userInner}>
            <View style={styles.gravatar}>
              <Image
                style={{ borderRadius: 5, width: 50, height: 50, paddingHorizontal: 10 }}
                source={{ uri: gravatar.avatarForEmail(user.email) }}
              />
            </View>
            <Text
              onPress={this.openPrompt}
              style={styles.inputInstructions}
            >{user.displayName}
            </Text>
          </View>
        </View>
        <Text style={styles.usernameHeader}>User Name</Text>
        <Text style={styles.username}>{user.username}</Text>
        <Text style={styles.emailHeader}>Email Address</Text>
        <Text style={styles.email}>{user.email}</Text>
      </Container>
    );
  }
}

Profile.propTypes = {
  loading: PropTypes.bool,
  onChangeName: PropTypes.func,
  user: PropTypes.shape({
    username: PropTypes.string.isRequired,
    email: PropTypes.string,
    displayName: PropTypes.string.isRequired,
  }),
};

export default Profile;
