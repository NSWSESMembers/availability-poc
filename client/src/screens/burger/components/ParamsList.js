import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { FlatList } from 'react-native';
import ParamsListItem from './ParamsListItem';


class ParamsList extends Component {
  renderItem = ({ item }) => (
    <ParamsListItem
      {...item}
    />
  )

  render() {
    const {
      deviceInfo,
      packageInfo,
      user,
      device,
      auth,
    } = this.props;

    const items = [
      {
        title: 'Device Info',
        detail: deviceInfo,
      },
      {
        title: 'Package Info',
        detail: packageInfo,
      },
      {
        title: 'Current auth state',
        detail: auth,
      },
      {
        title: 'Current device query',
        detail: device,
      },
      {
        title: 'Current user query',
        detail: user,
      },
    ];

    return (
      <FlatList
        data={items}
        keyExtractor={item => item.title}
        renderItem={this.renderItem}
      />
    );
  }
}

ParamsList.propTypes = {
  deviceInfo: PropTypes.string,
  packageInfo: PropTypes.string,
  auth: PropTypes.string,
  user: PropTypes.string,
  device: PropTypes.string,
};

export default ParamsList;
