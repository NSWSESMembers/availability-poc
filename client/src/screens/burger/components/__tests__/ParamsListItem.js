import React from 'react';
import Renderer from 'react-test-renderer';
import { Clipboard } from 'react-native';
import { ListItem } from '../../../../components/List';

import ParamsListItem from '../ParamsListItem';

const noop = () => {};

test('renders correctly', () => {
  const tree = Renderer.create(
    <ParamsListItem
      title="title"
      detail="detail"
      onPress={noop}
    />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('onPress', () => {
  const detail = 'detail';

  // mock the setString function
  Clipboard.setString = jest.fn();

  const renderer = Renderer.create(
    <ParamsListItem
      title="title"
      detail={detail}
      onPress={noop}
    />,
  );

  renderer.root.findByType(ListItem).props.onPress();

  expect(Clipboard.setString.mock.calls.length).toBe(1);
  expect(Clipboard.setString.mock.calls[0][0]).toBe(detail);
});
