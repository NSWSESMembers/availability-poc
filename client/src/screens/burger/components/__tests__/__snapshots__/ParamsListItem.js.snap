// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "borderRadius": 2,
          "elevation": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 0,
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 6,
          "paddingBottom": 10,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 10,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Object {
            "flex": 0,
            "fontWeight": "bold",
            "includeFontPadding": false,
            "marginRight": 8,
            "textAlignVertical": "top",
          }
        }
      >
        title
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Object {
            "color": "#6B6B6B",
            "flex": 0,
            "fontSize": 12,
            "includeFontPadding": false,
            "textAlignVertical": "bottom",
          }
        }
      >
        detail
      </Text>
    </View>
  </View>
</View>
`;
